osu.Game/Screens/Play/PauseContainer.cs

```diff
@@ -126,14 +126,52 @@ public class PauseOverlay : MenuOverlay
             public override string Header => "paused";
             public override string Description => "you're not going to do what i think you're going to do, are ya?";
 
+            private int selectedButton;
+
             protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
             {
                 if (!args.Repeat && args.Key == Key.Escape)
                 {
                     Buttons.Children.First().TriggerOnClick();
                     return true;
                 }
-
+                if (!args.Repeat && args.Key == Key.Up)
+                {
+                    if(selectedButton > 0)
+                    {
+                        Buttons[selectedButton].IsSelected = false;
+                        selectedButton--;
+                        Buttons[selectedButton].IsSelected = true;
+                    }
+                    return true;
+                }
+                if (!args.Repeat && args.Key == Key.Down)

```

Comment:
How about moving this to the base MenuOverlay class? Reusing code for both FailOverlay and PauseOverlay.

Reply:
I'm Not sure your idea.
I think that MenuOverlay Class does not responsibility that one selection menu.
So, I think a new class named by SelectionMenuOverlay

how do you think .

Reply:
Why not? MenuOverlay is used solely for these two overlays.

Reply:
Umm, Well, that means Menuoverlay class will not used others?

Reply:
Well it's in the Screens.Play namespace right now, so I believe that is a fair assumption. It's very specialised to gameplay right now - there's a separate dialog type for the rest of the game.

Reply:
Umm. I see. ok I will try this.